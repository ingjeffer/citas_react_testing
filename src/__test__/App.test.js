// import { render, screen } from '@testing-library/react';
// import App from './App';

import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "../App";

import axios from 'axios';

jest.mock('axios');


// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

test('<App /> La aplicaciÃ³n funciona bien la primera vez', () => {
    // const wrapper = render(<App />);
    // wrapper.debug();
    render(<App />);

    expect(screen.getByText('Administrador de Pacientes')).toBeInTheDocument();
    expect(screen.getByTestId('nombre-app').textContent).toBe('Administrador de Pacientes')
    expect(screen.getByTestId('nombre-app').tagName).toBe('H1')
    
    expect(screen.getByText('No hay citas')).toBeInTheDocument();
    expect(screen.getByText('Crear Cita')).toBeInTheDocument();
});

test('<App /> Valida agregar cita', () => {
    render(<App />);

    fireEvent.change(screen.getByTestId('mascota'), {
        target: {
            value: 'Hook',
        },
    });
    fireEvent.change(screen.getByTestId('propietario'), {
        target: {
            value: 'Juan',
        },
    });

    
    userEvent.type(screen.getByTestId('sintomas'), 'Fiebre');
    userEvent.type(screen.getByTestId('fecha'), '2022-03-22');
    userEvent.type(screen.getByTestId('hora'), '12:00');


    const btnSubmit = screen.getByTestId('btn-submit');
    // fireEvent.click(btnSubmit);
    userEvent.click(btnSubmit);

    // Revisar la alerta
    const alerta = screen.queryByTestId('alerta'); // queryByTestId: el elemento que consulta puede o no existir
    expect(alerta).not.toBeInTheDocument();


    // Revisar por el titulo dinamico
    expect(screen.getByTestId('titulo-dinamico').textContent).toBe('Administra tus Citas');
    expect(screen.getByTestId('titulo-dinamico').textContent).not.toBe('No hay citas');
});

test('<App /> Verificar las citas en el DOM', async () => {
    render(<App />);

    const citas = await screen.findAllByTestId('cita');
    console.log(citas);


    expect(citas).toMatchSnapshot();
});


test('Simula axios', () => {
    const data = {
        data: {
          hits: [
            {
              objectID: '1',
              title: 'a',
            },
            {
              objectID: '2',
              title: 'b',
            },
          ],
        },
      };
  
    axios.get.mockImplementationOnce(() => Promise.resolve(data));
})