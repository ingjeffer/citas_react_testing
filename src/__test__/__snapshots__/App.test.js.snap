// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Verificar las citas en el DOM 1`] = `
Array [
  <div
    class="cita"
    data-testid="cita"
  >
    <p>
      Mascota: 
      <span>
        Hook
      </span>
       
    </p>
    <p>
      Dueño: 
      <span>
        Juan
      </span>
       
    </p>
    <p>
      Fecha: 
      <span>
        2022-03-22
      </span>
       
    </p>
    <p>
      Hora: 
      <span>
        12:00
      </span>
       
    </p>
    <p>
      Sintomas: 
      <span>
        Fiebre
      </span>
       
    </p>
    <button
      class="button eliminar u-full-width"
    >
      Eliminar ×
    </button>
  </div>,
]
`;
